### PR TITLE
Fix orphaned browser processes due to uncaught exceptions in the tests

### PR DESCRIPTION
**Pull request by timvandermeij**
If uncaught exceptions occur in the tests (which happened in #17962 and can be triggered manually by throwing an error in `integration-boot.js`) the teardown logic of the tests doesn't get to run and thus spawned browser processes are not closed properly. Given that `test.mjs` is the only process that has a reference to them they will become orphaned and keep running if `test.mjs` exits without explicitly closing them.

This commit fixes the issue by always closing the browsers if uncaught exceptions occur, and we make sure to log them for debugging purposes.

Unblocks #17962.

_This patch is ever so slightly easier to review with the `?w=1` flag: https://github.com/mozilla/pdf.js/pull/18401/files?w=1._

### CODE REVIEW

**Comment by timvandermeij** [comment]
I have applied the following diff:

```diff
diff --git a/test/integration-boot.mjs b/test/integration-boot.mjs
index 4e04bb6f1..332c97a50 100644
--- a/test/integration-boot.mjs
+++ b/test/integration-boot.mjs
@@ -20,6 +20,8 @@ async function runTests(results) {
   jasmine.exitOnCompletion = false;
   jasmine.jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;
 
+  throw new Error("Yikes, something unexpected happened");
+
   jasmine.loadConfig({
     random: false,
     spec_dir: "integration"
```

Before this patch `npx gulp integrationtest` would crash, keep the Chrome process open and log this:

```
file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration-boot.mjs:23
  throw new Error("Yikes, something unexpected happened");
        ^

Error: Yikes, something unexpected happened
    at runTests (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration-boot.mjs:23:9)
    at startIntegrationTest (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/test.mjs:818:9)

Node.js v20.15.0
[12:46:24] The following tasks did not complete: integrationtest, runIntegrationTest
Did you forget to signal async completion?
file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/gulpfile.mjs:723
        throw new Error(`Running ${testsName} tests failed.`);
        ^

Error: Running integration tests failed.
    at ChildProcess.<anonymous> (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/gulpfile.mjs:723:15)
    at ChildProcess.emit (node:events:519:28)
    at ChildProcess.emit (node:domain:551:15)
    at maybeClose (node:internal/child_process:1105:16)
    at ChildProcess._handle.onexit (node:internal/child_process:305:5)
    at Process.callbackTrampoline (node:internal/async_hooks:130:17)

Node.js v20.15.0
```

After this patch `npx gulp integrationtest` exits normally, closes all browser processes and logs this:

```
Error: Yikes, something unexpected happened
    at runTests (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/integration-boot.mjs:23:9)
    at startIntegrationTest (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/test.mjs:818:9)
    at async main (file:///home/timvandermeij/Documenten/Ontwikkeling/pdf.js/Code/test/test.mjs:1093:7)

Run 0 tests
OHNOES!  No integration tests ran!
integration tests runtime was 3.3 seconds
```

Not only does this therefore fix the problem of hanging browser windows, but it also makes the logs more concise/actionable by removing noise from the uncaught exception propagating.

File: test/test.mjs [hunk 1]
```diff
@@ -1078,25 +1078,33 @@ async function main() {
...
+    // processes can become orphaned and keep running after `test.mjs` exits
+    // because the teardown logic of the tests did not get a chance to run.
+    console.error(e);
+    await Promise.all(sessions.map(session => closeSession(session.name)));
   }
 }
 
 var server;
-var sessions;
+var sessions = [];
```
Comment (Snuffleupagus) [inline test/test.mjs:1107]: Given this change, can the following line be removed now?
https://github.com/mozilla/pdf.js/blob/db9115625b9e16b79ffdfcd8481547ae6f05a90a/test/test.mjs#L974

Comment (timvandermeij) [reply test/test.mjs:1105]: Good catch; yes, that's possible now. I have removed it in the new commit.

**Review by Snuffleupagus — APPROVED** [review]
r=me, thank you!

**Pull request merged**